src/main/routing/shd-topology.c

```diff
@@ -122,6 +122,71 @@ static gboolean _topology_loadGraph(Topology* top, const gchar* graphPath) {
     return TRUE;
 }
 
+/** Returns FALSE if issue parsing graph, otherwise returns TRUE.
+ * If returning FALSE, consider result to be undefined.
+ * If returning TRUE, whether or not the graph is complete is stored in result.
+ */
+static gboolean _topology_isComplete(Topology* top, gboolean *result) {
+    MAGIC_ASSERT(top);
+    g_assert(result);
+    igraph_t *graph = &top->graph;
+    igraph_vs_t vs;
+    igraph_vit_t vit;
+    int ret;
+    igraph_integer_t vcount = igraph_vcount(graph);
+    gboolean is_success = FALSE;
+    gboolean is_complete = FALSE;
+    /*
+     * Determines if a graph is complete by:
+     * - knowning how many vertexes there are
+     * - for each vertex, count the indcident edges
+     *   - if less than the number of vertexes, it isn't a complete graph
+     * - otherwise the graph is complete
+    /* vert selector. We wall all verts */
+    ret = igraph_vs_all(&vs);
+    if (ret != IGRAPH_SUCCESS) {
+        critical("igraph_vs_all returned non-success code %i", ret);
+        is_success = FALSE;
+        goto done;
+    }
+    ret = igraph_vit_create(graph, vs, &vit);
+    if (ret != IGRAPH_SUCCESS) {
+        critical("igraph_vit_create returned non-success code %i", ret);
+        is_success = FALSE;
+        goto done;
+    }
+    while (!IGRAPH_VIT_END(vit)) {
+        igraph_vector_t iedges;
+        igraph_vector_init(&iedges, 0);
+        ret = igraph_incident(graph, &iedges, IGRAPH_VIT_GET(vit), IGRAPH_OUT);
+        if (ret != IGRAPH_SUCCESS) {
+            critical("error computing igraph_incident\n");
+            is_success = FALSE;
+            igraph_vector_destroy(&iedges);
+            goto done;
+        }
+        igraph_integer_t ecount = igraph_vector_size(&iedges);
+        if (ecount < vcount) {
+            info("Vert id=%li has %li incident edges to %li total verts "
+                "and thus this isn't a complete graph",
+                IGRAPH_VIT_GET(vit), ecount, vcount);
```

Comment:
Since you changed `ecount` and `vcount` types to `igraph_integer_t`, you need to cast them to `long int` here in the `info()` log function. (We can't guarantee that `igraph_integer_t` is defined as a `long int`everywhere, and since you are using the `%li` format string, you need to cast.)

It would be better if igraph provided something like an `IGRAPH_INTEGER_FORMAT` macro that always inserted the correct format string, but I wasn't able to find it.